modules/base/rotation/timelinerotation.cpp

```diff
@@ -0,0 +1,114 @@
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rotation/timelinerotation.h>
+
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/util/updatestructures.h>
+#include <openspace/util/time.h>
+
+namespace {
+    constexpr const char* KeyType = "Type";
+    constexpr const char* KeyKeyframes = "Keyframes";
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation TimelineRotation::Documentation() {
+    using namespace documentation;
+    return {
+        "Timeline Rotation",
+        "base_transform_rotation_keyframe",
+        {
+            {
+                KeyType,
+                new StringEqualVerifier("TimelineRotation"),
+                Optional::No
+            },
+            {
+                KeyKeyframes,
+                new TableVerifier({
+                    { "*", new TableVerifier(), Optional::No, "Any translation object" }
+                }),
+                Optional::No,
+                "A table of keyframes, with keys formatted as YYYY-MM-DDTHH:MM:SS"
+                "and values that are valid Rotation objects."
+            }
+        }
+    };
+}
+
+TimelineRotation::TimelineRotation(const ghoul::Dictionary& dictionary) {
+    documentation::testSpecificationAndThrow(
+        Documentation(),
+        dictionary,
+        "TimelineTranslation"
+    );
+
+    const ghoul::Dictionary& keyframes =
+        dictionary.value<ghoul::Dictionary>(KeyKeyframes);
+
+    std::vector<std::string> timeStrings = keyframes.keys();
+    for (const std::string& timeString : timeStrings) {
+        const Time t = Time::convertTime(timeString);
+
+        std::unique_ptr<Rotation> rotation = 
+            Rotation::createFromDictionary(
+                keyframes.value<ghoul::Dictionary>(timeString)
+            );
+
+        if (rotation) {
+            _timeline.addKeyframe(t.j2000Seconds(), std::move(rotation));
+        }
+    }
+}
+
+glm::dmat3 TimelineRotation::matrix(const UpdateData& data) const {
+    const double now = data.time.j2000Seconds();
+    using KeyframePointer = const Keyframe<std::unique_ptr<Rotation>>*;
+
+    KeyframePointer prev = _timeline.lastKeyframeBefore(now, true);
+    KeyframePointer next = _timeline.firstKeyframeAfter(now, true);
+
+    if (!prev && !next) {
+        return glm::mat3(0.0);
+    }
+    if (!prev) {
+        prev = next;
+    }
+    if (!next) {
+        next = prev;
+    }
+    const double prevTime = prev->timestamp;
+    const double nextTime = next->timestamp;
+
+    double t = 0.0;
+    if (nextTime - prevTime > 0.0) {
+        t = (now - prevTime) / (nextTime - prevTime);
+    }
+    return t * next->data->matrix(data) + (1.0 - t) * prev->data->matrix(data);
```

Comment:
I think here we should use a slerp instead of a lerp. Just interpolating the individual components of a matrix can give wierd results, which will probably be visible if the rotation between two keyframes is big enough.